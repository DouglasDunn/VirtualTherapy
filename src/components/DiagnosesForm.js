import React from 'react';

export default class DiagnosesForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      diagnoses: '',
      medication: '',
      time: ''
    };
  }

  onDiagnosesChange = (e) => {
    const diagnoses = e.target.value;
    this.setState(() => ({ diagnoses }));
  };

  onMedicationChange = (e) => {
    const medication = e.target.value;
    this.setState(() => ({ medication }));
  };

  onTimeChange = (e) => {
    const time = e.target.value;
    this.setState(() => ({ time }));
  };

  onSubmit = (e) => {
    e.preventDefault();

    // this.props.onSubmit({
    //   name: this.state.name,
    //   age: this.state.age,
    //   emergencyContactName: this.state.emergencyContactName,
    //   emergencyContactNumber: this.state.emergencyContactNumber,
    // });
  };

  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        <input
          type="text"
          placeholder="Diagnoses or Symptom"
          autoFocus
          className="text-input"
          value={this.state.diagnoses}
          onChange={this.onDiagnosesChange}
        />
        <input
          type="text"
          placeholder="Medication"
          className="text-input"
          value={this.state.medication}
          onChange={this.onMedicationChange}
        />
        <input
          type="text"
          placeholder="Time"
          className="text-input"
          value={this.state.time}
          onChange={this.onTimeChange}
        />
        <div>
          <button className="button">Submit</button>
        </div>
      </form>
    )
  }
}
