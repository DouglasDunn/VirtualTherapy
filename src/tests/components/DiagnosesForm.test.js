import React from 'react';
import { shallow } from 'enzyme';
import DiagnosesForm from '../../components/DiagnosesForm';
// import profile from '../fixtures/profile';

test('should render DiagnosesForm correctly', () => {
  const wrapper = shallow(<DiagnosesForm />);
  expect(wrapper).toMatchSnapshot();
});

test('should set diagnoses on input change', () => {
  const value = 'New diagnoses';
  const wrapper = shallow(<DiagnosesForm />);
  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('diagnoses')).toBe(value);
});

test('should set medication on input change', () => {
  const value = 'New medication';
  const wrapper = shallow(<DiagnosesForm />);
  wrapper.find('input').at(1).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('medication')).toBe(value);
});

test('should set time on input change', () => {
  const value = 'New time';
  const wrapper = shallow(<DiagnosesForm />);
  wrapper.find('input').at(2).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('time')).toBe(value);
});
