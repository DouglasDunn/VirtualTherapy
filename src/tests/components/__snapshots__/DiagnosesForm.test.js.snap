// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DiagnosesForm correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Diagnoses or Symptom"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Medication"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <div>
    <button
      className="button"
    >
      Submit
    </button>
  </div>
</form>
`;
